scanamo/src/test/scala/org/scanamo/TableTest.scala

```diff
@@ -593,4 +592,46 @@ class TableTest extends AnyFunSpec with Matchers {
       )
     }
   }
+
+  it("Filters a table with a very long condition expression") {
+    LocalDynamoDB.withRandomTable(client)("line" -> S, "name" -> S) { t =>
+      val stationTable = Table[Station](t)
+      val ops = for {
+        _ <- stationTable.putAll(
+          Set(
+            Station("Metropolitan", "Chalfont & Latimer", 8),
+            Station("Metropolitan", "Chorleywood", 7),
+            Station("Metropolitan", "Rickmansworth", 7),
+            Station("Metropolitan", "Croxley", 7),
+            Station("Jubilee", "Canons Park", 5)
+          )
+        )
+        filteredStations <-
+          stationTable
+            .filter(
+              (
+                attributeExists("line") &&
+                  attributeExists("name") &&
+                  attributeNotExists("colour") &&
+                  !attributeExists("speed") &&
+                  "zone" -> Set(8, 7, 5) &&
+                  ("line" beginsWith "Metr") &&
+                  ("name" beginsWith "C") &&
+                  ("zone" between (5 and 8)) &&
```

Comment:
I think the symbolic representation and operational semantics are orthogonal. In the same table there are languages using `and` and `or` instead of `&&` and `||` that short-circuit evaluation. But you are right that no one asked for this, it is purely because the `and` operator is overloaded in many different places with different meanings, the two notable ones are:

- combining condition expressions in `cond1 and cond2`
- defining bounds in `between x and y`

I'll revert that change, it is not related to solving this bug.

Reply:
Fixed.